pkgs/applications/misc/k40-whisperer/default.nix

```diff
@@ -46,14 +48,29 @@ in stdenv.mkDerivation rec {
 
     mkdir -p $out/bin
     mkdir -p $out/lib/udev/rules.d
+    mkdir -p $out/share/icons/hicolor/128x128/apps
+    mkdir -p "$out/share/applications"
 
     ln -s ${udevRule} $out/lib/udev/rules.d/97-k40-whisperer.rules
+    ln -s "$desktopItem/share/applications/k40whisperer.desktop" "$out/share/applications/k40whisperer.desktop"
+
+    gm convert "$out/scorchworks.ico" "$out/share/icons/hicolor/128x128/apps/scorchworks.png"
 
     makeWrapper '${pythonEnv.interpreter}' $out/bin/k40-whisperer \
       --add-flags $out/k40_whisperer.py \
       --prefix PYTHONPATH : $out
   '';
 
+  desktopItem = makeDesktopItem {
```

Comment:
Move this into a let binding please. I know you can put anything into `mkDerivation` and it will magically show up as shell variable in the build script, however this is horrible and should never have been possible in the first place and I really hope we'll get a sane mkDerivation one day …